ffs/filcold/filcold.go

```diff
@@ -77,50 +81,80 @@ func (fc *FilCold) Fetch(ctx context.Context, pyCid cid.Cid, piCid *cid.Cid, wad
 	return ffs.FetchInfo{RetrievedMiner: miner, FundsSpent: fundsSpent}, nil
 }
 
+func (fc *FilCold) calculatePieceSize(ctx context.Context, c cid.Cid) (api.DataSize, error) {
```

Comment:
The two places that need to calcualte the piece size now:
- Use the Lotus API instead of our custom logic.
- Are rate limited by the deal-preparation-limit flag.
- Even if they pass the rate-limiting, they're blocked if the Lotus sync monitor is claiming the lotus node is having a hard time syncing, so it can recovers.